src/praisonai-train/praisonai_train/data/generate.py

```
@@ -0,0 +1,121 @@
+    }
+    headers = {"Content-Type": "application/json"}
+    if cfg.get("azure"):
+        headers["api-key"] = cfg["api_key"]
+    else:
+        headers["Authorization"] = f"Bearer {cfg['api_key']}"
+        body["model"] = cfg["deployment"]
+    req = urllib.request.Request(url, data=json.dumps(body).encode(), headers=headers)
+    content = None
+    for attempt in range(2):
+        try:
+            with urllib.request.urlopen(req, timeout=120) as resp:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)